qiskit_ibm/hgp.py

```diff
@@ -0,0 +1,106 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A hub, group and project in an IBM Quantum account."""
+
+import logging
+from collections import OrderedDict
+import traceback
+from typing import Any, Dict, Optional
+
+from qiskit.providers.models import PulseBackendConfiguration, QasmBackendConfiguration
+from .utils.json_decoder import decode_backend_configuration
+from .ibm_backend import IBMBackend, IBMSimulator
+from .api.clients import AccountClient
+from .credentials import Credentials
+
+logger = logging.getLogger(__name__)
+
+
+class HGP:
+    """Provides access to IBM Quantum backends associated with a hub, group and project"""
```

Comment:
I can improve the documentation. Do you have any suggestions for a different name? I don't really care what the name is since it's a throwaway concept. (I am not really good at naming things lol)

Reply:
Maybe `BackendGroup` (grouping of backends based on hgp) or `HBPBackends` (backends that belong to a hgp) or `ProjectBackends`?